### PR TITLE
feat(warehouse): timeout during warehouse table count

### PR DESCRIPTION
**Pull request by achettyiitr**
# Description

Added timeouts for warehouse table upload count. This is required because we have seen sometimes queries getting struck and running for minutes which is essentially not required.

## Notion Ticket

https://www.notion.so/rudderstacks/Add-timeout-for-data-quality-checks-like-count-Or-getting-count-of-rows-after-sync-from-output-of-76c232dba33d47b6b55d8d2bb476f7ee

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: warehouse/warehouse.go [hunk 1]
```diff
@@ -77,6 +77,7 @@ var (
 	pkgLogger                           logger.Logger
 	numLoadFileUploadWorkers            int
 	slaveUploadTimeout                  time.Duration
+	totalCountInTableTimeout            time.Duration
```
Comment (achettyiitr) [inline warehouse/warehouse.go:80]: Change variable.

File: warehouse/warehouse.go [hunk 2]
```diff
@@ -189,6 +190,8 @@ func loadConfig() {
 	config.RegisterBoolConfigVariable(false, &skipDeepEqualSchemas, true, "Warehouse.skipDeepEqualSchemas")
 	config.RegisterIntConfigVariable(8, &maxParallelJobCreation, true, 1, "Warehouse.maxParallelJobCreation")
 	config.RegisterBoolConfigVariable(false, &enableJitterForSyncs, true, "Warehouse.enableJitterForSyncs")
+	config.RegisterDurationConfigVariable(30, &totalCountInTableTimeout, true, time.Second, []string{"Warehouse.totalCountInTableTimeout", "Warehouse.totalCountInTableTimeoutInSec"}...)
```
Comment (achettyiitr) [inline warehouse/warehouse.go:193]: InS

Comment (achettyiitr) [reply warehouse/warehouse.go:193]: tableCountTimeout

Comment (Jayachand) [reply warehouse/warehouse.go:193]: or tableCountQueryTimeout

**Review by Jayachand — APPROVED** [review]

File: warehouse/bigquery/bigquery.go [hunk 3]
```diff
@@ -775,7 +775,7 @@ func (bq *HandleT) IsEmpty(warehouse warehouseutils.Warehouse) (empty bool, err
 		if !exists {
 			continue
 		}
-		count, err := bq.GetTotalCountInTable(tableName)
+		count, err := bq.GetTotalCountInTable(bq.BQContext, tableName)
```
Comment (achettyiitr) [inline warehouse/bigquery/bigquery.go:778]: Make it background context.

Comment (achettyiitr) [reply warehouse/bigquery/bigquery.go:778]: Will make this as part of separate PR.

**Pull request merged**